actions/build/action.yml

```diff
@@ -0,0 +1,155 @@
+      API_TOKEN=${{ secrets.API_TOKEN }}
+      ```
+    required: false
+    default: ""
+  build-artifact:
+    description: |
+      JSON object specifying artifact upload configuration.
+      Format: {"name": "artifact-name", "paths": "path1\npath2"}
+    required: false
+    default: ""
+  container:
+    description: "Whether running in container mode (skips checkout and node setup)"
+    required: false
+    default: "false"
+
+outputs:
+  artifact-id:
+    description: "ID of the uploaded artifact (if artifact was specified)"
+    value: ${{ steps.build-artifact-id.outputs.artifact-id }}
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      # FIXME: workaround until will be merged: https://github.com/actions/runner/pull/1684
+      run: mkdir -p ./self-build-action/ && cp -r $GITHUB_ACTION_PATH/../* ./self-build-action/
+
+    - id: setup-node
+      if: inputs.container != 'true'
+      uses: ./self-build-action/setup-node
+      with:
+        working-directory: ${{ inputs.working-directory }}
+        dependencies-cache: |
+          nx
+          gatsby
+          storybook
+
+    - id: get-package-manager
+      if: inputs.container == 'true'
+      uses: ./self-build-action/get-package-manager
+      with:
+        working-directory: ${{ inputs.working-directory }}
+
+    - uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      env:
+        BUILD_ENV: ${{ inputs.build-env }}
+        BUILD_SECRETS: ${{ inputs.build-secrets }}
+      with:
+        script: |
+          const envInput = process.env.BUILD_ENV || '{}';
+
+          let buildEnv = {};
+
+          try {
+            buildEnv = JSON.parse(envInput);
+          } catch (e) {
+            core.setFailed(`Invalid build env JSON: ${e.message}`);
+          }
+
+          for (const [key, value] of Object.entries(buildEnv)) {
+            core.exportVariable(key, value);
+          }
+
+          const secretsInput = process.env.BUILD_SECRETS || '';
+          for (const line of secretsInput.split('\n').map(line => line.trim()).filter(Boolean)) {
+            const [key, ...rest] = line.split('=');
+            if (!key || !rest.length) {
+              return core.setFailed(`Invalid build secrets format: ${line}`);
+            }
+            const value = rest.join('=');
+            core.exportVariable(key.trim(), value.trim());
+          }
+
+    # jscpd:ignore-start
+    - uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      env:
+        BUILD_COMMANDS: ${{ inputs.build-commands }}
+        RUN_SCRIPT_COMMAND: ${{ inputs.container == 'true' && steps.get-package-manager.outputs.run-script-command || steps.setup-node.outputs.run-script-command }}
+        WORKING_DIRECTORY: ${{ inputs.working-directory }}
+      with:
+        script: |
+          const { exec } = require('child_process');
+          const { promisify } = require('util');
+          const execAsync = promisify(exec);
```

Comment:
Use available exec package instead of using child_process (https://github.com/actions/github-script?tab=readme-ov-file#using-exec-package)